ott/tools/segment_sinkhorn.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2022 The OTT Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Segmented sinkhorn utility."""
+
+from typing import Any, Mapping, Optional
+
+from jax import numpy as jnp
+
+from ott.core import segment, sinkhorn
+from ott.geometry import costs, pointcloud
+
+
+def segment_sinkhorn(
+    x: jnp.ndarray,
+    y: jnp.ndarray,
+    cost_fn: Optional[costs.CostFn] = None,
+    segment_ids_x: Optional[jnp.ndarray] = None,
+    segment_ids_y: Optional[jnp.ndarray] = None,
+    num_segments: Optional[int] = None,
+    indices_are_sorted: Optional[bool] = None,
+    num_per_segment_x: Optional[jnp.ndarray] = None,
+    num_per_segment_y: Optional[jnp.ndarray] = None,
+    weights_x: Optional[jnp.ndarray] = None,
+    weights_y: Optional[jnp.ndarray] = None,
+    sinkhorn_kwargs: Optional[Mapping[str, Any]] = None,
```

Comment:
I slightly prefer `sinkhorn_kwargs: Mapping[str, Any] = MappingProxyType({})`, but it's minor and I do it in #116

Reply:
Thanks!